Fix integration tests

kalilsn opened the pull request:
Preliminary work for https://github.com/department-of-veterans-affairs/vets-contrib/issues/2492

The integration tests are now runnable with `npm run test-integration`. The snapshots can be updated with `npm run test-integration -- -u`. You'll need a local kong instance running on port 8001, which I was able to achieve with https://github.com/Kong/docker-kong/tree/master/compose. In order to run this with 0.14.1, I first had to change the `kong-migrations` service's command to `kong migrations up`, then ran `KONG_DOCKER_TAG=kong:0.14.1 docker-compose up`.

Commit:
Make integration tests runnable

Commit:
Run integration tests serially

Commit:
Fix add SNI request

Kong 0.14 changed the format of the add SNI payload from
{ "name": "example.com", "ssl_certificate_id": "21b69eab-09d9-40f9-a55e-c4ee47fada68" }
to
{ "name": "example.com", "certificate": { "id": "21b69eab-09d9-40f9-a55e-c4ee47fada68" } }

Commit:
Delete redundant integration tests

These tests are the same as the "config" tests, but they include the
config inline as a JS object. The config tests are preferable because
they also test the yaml format, which we use.

Commit:
Increase jest timeout

Kongfig retries failed requests a few times, which takes longer than
the default 5 second timeout. When that happens, the test output hides
the actual error messages from kong and instead shows a timeout error.

Commit:
Include kong host in npm script

Commit:
Prevent error on null log message

Commit:
Fix second sni payload bug

Commit:
Update snapshots and delete obsolete ones

Removes all of the snapshots associated with removed tests.
Changes within the snapshots are the result of the SNI object changes
and the addition of the consumer.custom_id field

dvogel commented:
Is the `ssl_certificate` -> `certificate` change breaking backward compatibility?

kalilsn commented:
> Is the ssl_certificate -> certificate change breaking backward compatibility?

Yes, it's breaking compatibility with 0.13 and earlier, in order to fix compatibility with 0.14 onwards, which we're currently using.

dvogel reviewed (changes requested):
> > Is the ssl_certificate -> certificate change breaking backward compatibility?
> 
> Yes, it's breaking compatibility with 0.13 and earlier, in order to fix compatibility with 0.14 onwards, which we're currently using.

:+1: Can you make sure this change in version compatibility is reflected in the README?

Commit:
Add compatibility notice to readme

Commit:
Run pretest script before integration tests

npm automatically runs the pretest script before the test script, so
this explicitly ensures the same thing for the integration tests.

Commit:
Revert "Run pretest script before integration tests"

This reverts commit c333f5d01d0145f13900ec618cd3c6f31bd55788.

Commit:
Import from uncompiled files in tests for easier debugging
